### PR TITLE
[Merged by Bors] - Update `Query` methods documentation

### LINKED ISSUE #5506: The `iter_many_mut` documentation is confused
> Calls a closure on each result of [Query](https://docs.rs/bevy/latest/bevy/ecs/system/struct.Query.html) where the entities match.

The documentation talks about calling a "closure", but `iter_many_mut` doesn't have a closure argument, as the example shows. https://docs.rs/bevy/latest/bevy/ecs/system/struct.Query.html#method.iter_many_mut

### PR DESCRIPTION
**Pull request by Nilirad**
# Objective

- Increase consistency across documentation of `Query` methods.
- Fixes #5506

## Solution

- See #4989. This PR is derived from it. It just includes changes to the `Query` methods' docs.

### CODE REVIEW

File: crates/bevy_ecs/src/system/query.rs [hunk 1]
```diff
@@ -576,15 +605,11 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
         )
     }
 
-    /// Runs `f` on each query result. This is faster than the equivalent iter() method, but cannot
-    /// be chained like a normal [`Iterator`].
-    ///
-    /// This can only pass in immutable data, see [`Self::for_each_mut`] for mutable access.
+    /// Runs `f` on each read-only query item.
```
Comment (Weibye) [inline crates/bevy_ecs/src/system/query.rs:608]: Is there value in keeping the note on "faster than iter()"?

File: crates/bevy_ecs/src/system/query.rs [hunk 2]
```diff
@@ -1085,11 +1150,11 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
...
 
-    /// Returns a single immutable query result when there is exactly one entity matching
-    /// the query.
+    /// Returns a single read-only query item when there is exactly one entity matching the query.
+    ///
+    /// # Panics
     ///
-    /// This can only return immutable data. Use [`single_mut`](Self::single_mut) for
-    /// queries that contain at least one mutable component.
+    /// This method panics if the number of query items is not exactly one.
```
Comment (Weibye) [inline crates/bevy_ecs/src/system/query.rs:1157]:
```suggestion
    /// This method panics if the number of query items is **not** exactly one.
```

File: crates/bevy_ecs/src/system/query.rs [hunk 3]
```diff
@@ -1160,8 +1225,11 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
         }
     }
 
-    /// Returns a single mutable query result when there is exactly one entity matching
-    /// the query.
+    /// Returns a single query item when there is exactly one entity matching the query.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the number of query item is not exactly one.
```
Comment (Weibye) [inline crates/bevy_ecs/src/system/query.rs:1232]:
```suggestion
    /// This method panics if the number of query item is **not** exactly one.
```

**Review by Weibye — APPROVED** [review]
This reads more clear and consistent, good work :)

File: crates/bevy_ecs/src/system/query.rs [hunk 4]
```diff
@@ -663,16 +695,16 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
...
-    ///* `f` - The function to run on each item in the query
+    /// [`ComputeTaskPool`]: bevy_tasks::prelude::ComputeTaskPool
     ///
     /// # Panics
-    /// The [`ComputeTaskPool`] is not initialized. If using this from a query that is being
-    /// initialized and run from the ECS scheduler, this should never panic.
     ///
-    /// [`ComputeTaskPool`]: bevy_tasks::prelude::ComputeTaskPool
+    /// This method panics if the [`ComputeTaskPool`] resource is added to the `World` before using this method.
+    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this should never panic.
```
Comment (TimJentzsch) [inline crates/bevy_ecs/src/system/query.rs:703]: I think we should either guarantee this or leave it out if we can't

```suggestion
    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this never panics.
```

File: crates/bevy_ecs/src/system/query.rs [hunk 5]
```diff
@@ -692,14 +724,20 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
...
+    /// Runs `f` on each read-only query item in parallel.
+    ///
+    /// Parallelization is achieved by using the [`World`]'s [`ComputeTaskPool`].
     ///
     /// # Panics
-    /// The [`ComputeTaskPool`] is not initialized. If using this from a query that is being
-    /// initialized and run from the ECS scheduler, this should never panic.
+    ///
+    /// This method panics if the [`ComputeTaskPool`] resource is added to the `World` before using this method.
+    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this should never panic.
```
Comment (TimJentzsch) [inline crates/bevy_ecs/src/system/query.rs:734]: Same as above, all or nothing :)

```suggestion
    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this never panics.
```

File: crates/bevy_ecs/src/system/query.rs (hunk 1 above)
Comment (Nilirad) [reply crates/bevy_ecs/src/system/query.rs:608]: I think not, #5741 has a dedicated performance section.

**Review by TimJentzsch — APPROVED** [review]
Nicely done :)

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**